acasclient/lsthing.py

```diff
@@ -1675,8 +1675,9 @@ class SimpleLsThing(BaseModel):
 
     ROW_NUM_KEY = 'row number'
 
-    def __init__(self, ls_type=None, ls_kind=None, code_name=None, names={}, ids={}, aliases={}, metadata={}, results={}, links=[], recorded_by=None,
-                 preferred_label_kind=None, state_tables=defaultdict(dict), ls_thing=None, client=None):
+    def __init__(self, ls_type=None, ls_kind=None, code_name=None, names=None, ids=None, aliases=None, metadata=None, results=None, links=None, recorded_by=None,
+                 preferred_label_kind=None, state_tables=None, ls_thing=None, client=None):
+        metadata = metadata or {}
```

Comment:
Confused why this is here and not on line 1704. If there is a reason it seems like `results` and `state_tables` should receive the same treatment.

Reply:
I did that to follow DRY
```
metadata = metadata or {}
foo = metadata
bar = fn(metadata)
```
vs 
```
foo = metadata or {}
bar = fn(metadata or {})
```
Two functions use metadata more than once, but others use it once only (hence direct modification for those).
I saw your commit, now below places will pick `None` as `metadata` value instead of `{}`
https://github.com/mcneilco/acasclient/blob/master/acasclient/lsthing.py#L2153
https://github.com/mcneilco/acasclient/blob/master/acasclient/lsthing.py#L1691

Reply:
I see. My mistake - sorry, and thanks for the fix.

Reply:
Sorry for the bad commit. Thanks for the fix.